### PR TITLE
feat(android): fetchUserInfo

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(android): fetchUserInfo

## Linear Issue Reference

* LOG-1102

## Testing

UT

### CODE REVIEW

File: android/logto-client/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt [hunk 1]
```diff
@@ -11,10 +11,12 @@ import io.logto.sdk.core.http.HttpCompletion
 import io.logto.sdk.core.type.IdTokenClaims
 import io.logto.sdk.core.type.OidcConfigResponse
 import io.logto.sdk.core.type.RefreshTokenTokenResponse
+import io.logto.sdk.core.type.UserInfoResponse
 import io.logto.sdk.core.util.TokenUtils
 import org.jetbrains.annotations.TestOnly
 import org.jose4j.jwt.consumer.InvalidJwtException
 
+@Suppress("TooManyFunctions")
```
Comment (gao-sun) [inline android/logto-client/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt:19]: 这个能不能分开定义啊？类似 swift 里的 extension

File: android/logto-client/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt [hunk 2]
```diff
@@ -139,6 +141,27 @@ open class LogtoClient(
...
+            requireNotNull(result).let { oidcConfig ->
+                getAccessToken { throwable: Throwable?, result: AccessToken? ->
+                    throwable?.let {
+                        callback.onResult(it, null)
+                        return@getAccessToken
+                    }
+                    Core.fetchUserInfo(
+                        oidcConfig.userinfoEndpoint,
+                        requireNotNull(result).token
+                    ) { throwable, response -> callback.onResult(throwable, response) }
```
Comment (gao-sun) [inline android/logto-client/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt:159]: can we directly assign `callback.onResult` to `fetchUserInfo` here

**Review by gao-sun — APPROVED** [review]

File: android/logto-client/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt (hunk 1 above)
Comment (xiaoyijun) [reply android/logto-client/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt:19]: WIP: LOG-1454

File: android/logto-client/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt (hunk 2 above)
Comment (xiaoyijun) [reply android/logto-client/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt]: updated

**Pull request merged**